src/cdomains/lval.ml

```diff
@@ -41,7 +41,7 @@ struct
     match x, y with
     | `NoOffset , `NoOffset -> true
     | `NoOffset, x
-    | x, `NoOffset -> cmp_zero_offset x = `MustZero (* cannot derive due to this special case *)
+    | x, `NoOffset -> cmp_zero_offset x = `MustZero (* cannot derive due to this special case, special cases not used for AddressDomain any more due to splitting *)
```

Comment:
This comment confuses me.

Reply:
If the special cases are no longer used, why still have them?

Reply:
Since I separated #816 from this PR, I wanted to minimize any other changes to the lvalue lattice as well because it needs a bigger refactoring/overhaul anyway.